std/algorithm/searching.d

```diff
@@ -2849,7 +2849,28 @@ true if the initial segment of $(D r1) matches $(D r2), and $(D r1) has been
 advanced to the point past this segment; otherwise false, and $(D r1) is left
 in its original position.
  */
-bool skipOver(alias pred = "a == b", R1, R2)(ref R1 r1, R2 r2)
+bool skipOver(R1, R2)(ref R1 r1, R2 r2)
+    if (isForwardRange!R1 && isInputRange!R2
+        && is(typeof(r1.front == r2.front)))
+{
+    static if (isSomeString!R1 && isSomeString!R2 &&
```

Comment:
Your string version would be better for things like int[] and float[] as well since I believe the meat of the function would be vectorized. Probably best to extend.